File: server/src/main/java/com/example/server/entity/Notification.java
package com.example.server.entity;

import com.example.server.entity.enums.NotificationStatus;
import com.example.server.entity.enums.NotificationType;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;

import java.util.UUID;

@Data
@ToString
@EqualsAndHashCode
@Document(indexName = "notification", shards = 1)
public class Notification {
    @Id
    private UUID id;
    private NotificationType type;
    private User from;
    private User to;
    private Project project;
    private NotificationStatus status;
    private String description;
}

File: client/src/app/table-classes/notification.ts
import {User} from "./user";
import {NotificationType} from "./notification-type";
import {NotificationStatus} from "./notification-status";
import {Project} from "./project";

export class Notification {
  constructor(public id: string, public type: NotificationType, public from: User, public to: User, public project: Project,
              public status: NotificationStatus, public description: string) {
  }
}
